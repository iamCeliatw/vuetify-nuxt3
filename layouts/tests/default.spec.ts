import { describe, it, expect } from "vitest";
import Default from "../default.vue";
import { mount } from "@vue/test-utils";
vi.stubGlobal("useRouter", () => ({}));
vi.stubGlobal("useSupabaseClient", () => ({}));
describe("Default layout", () => {
  it("should render correctly", () => {
    const wrapper = mount(Default);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("hamberger icon should render correctly", () => {
    const wrapper = mount(Default);
    // const drawer = wrapper.find(".v-navigation-drawer");
    const hamburgerIcon = wrapper.find(".icon-selector"); // 用你的實際選擇器替換 '.icon-selector'
    // expect(hamburgerIcon.classes()).toContain("mdi-hamburger");
  });
});
