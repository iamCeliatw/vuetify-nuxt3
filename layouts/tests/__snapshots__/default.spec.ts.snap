// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Default layout > should render correctly 1`] = `
"<v-layout data-v-433a9abd="" class="rounded rounded-md">
  <v-app-bar data-v-433a9abd="" color="admin-layout">
    <v-app-bar-nav-icon data-v-433a9abd="" icon="mdi-hamburger"></v-app-bar-nav-icon>
    <v-toolbar-title data-v-433a9abd="">Admin</v-toolbar-title>
    <v-spacer data-v-433a9abd=""></v-spacer>
    <v-btn data-v-433a9abd=""><img data-v-433a9abd="" src="/logout.png" alt=""></v-btn>
  </v-app-bar>
  <v-navigation-drawer data-v-433a9abd="" modelvalue="false">
    <v-divider data-v-433a9abd=""></v-divider>
    <v-list-item data-v-433a9abd="" class="list-item my-2" title="profile"></v-list-item>
    <v-list-item data-v-433a9abd="" class="list-item my-2" title="post"></v-list-item>
    <v-list-item data-v-433a9abd="" class="list-item my-2" title="category"></v-list-item>
    <v-list-item data-v-433a9abd="" class="list-item my-2" title="images"></v-list-item>
    <v-list-item data-v-433a9abd="" class="list-item my-2" title="tags"></v-list-item>
  </v-navigation-drawer>
  <v-main data-v-433a9abd=""></v-main>
</v-layout>"
`;
